### DISCUSSION TITLE
More absorbency

### DISCUSSION THREAD

**Discussion by Kerie70** [zippity21/Stardew_Valley_Regression_Mod#23] [Ideas]
Digging through the files a little I've noticed that the max absorbency is set to 2500 and only for 2 diapers.  Could we increase those to hold more or possibly create a new diaper all together with a higher rating?  (Little side question, can I change the rate in the file for my personal game without it causing an issue to the mod as a whole?)

**Comment by zippity21** [comment]
The idea with setting the numbers the way they are is that it reflects how much of the day it'll last you, with the best lasting for a whole day of wetting + 1 full mess, since I wouldn't expect a diaper to be used for multiple days. When I say "full," I mean when your bladder/bowels are at max capacity, since they shrink as you're continence decreases. (That being said, I actual do think the containment numbers need to be bumped slightly to contain the full mess) I sort of arbitrarily decided which diapers were "the best," but I didn't want to make them all equal.

Honestly though, I need opinions like these to make adjustments for how people really want to play.

You should be able to adjust the capacity of your current underwear in the RegressionSave.json file that gets created, but you do need to exit the game and reload for it to take effect. Though do so at your own risk! Similarly, you can edit en.json to adjust all the underwear options, but it won't effect the underwear you're currently in, so you'd need to change.

As for adding a new diaper: you can do it by just creating another entry in the en.json, exit the game and reload (it'll show up in the seed shop). However, you'll need to use an existing sprite index. So adding more is easy, but if we want to add a new design, we'd need to create a new sprite (which I am not artistic enough to do).

edit for grammer

**Reply by Kerie70** [reply]
Alright I think I got it.  If you could make a new discussion entry or type it here about how adding sprites work with the files I can take a crack at it.  I would love to create and add a pink(ish) diaper to the game.

**Reply by zippity21** [reply]
In the Asset folder there is a file called sprites.png. This file currently contains the sprites for 11 underwear (0-10 sprite index), arranged by column. There are 4 rows to each column; 1. Dry, 2. Wet (with pee), 3. Messy, 4. Wet (with water).

Any diaper added needs to be given a new column and all 4 rows need to be filled in. Since its a png file, you can edit it right in paint if you want.

The size requirement is that each sprite takes up EXACTLY a 16x16 square. Obviously, the pixels you don't use must be transparent.

Then all you need to do is set the sprite index variable in the diaper's entry in en.json to the new column (starting at 0).

**Reply by Kerie70** [reply]
Thank you for the info, I managed to pull it off both with the en.json edit and the sprite models.  Took me a hot minute to make the diapers but I came up with a relatively simple pink heart diaper and got all 4 sprites to work perfectly.  Even showing up in the shop and so on.  (Another side question.  Will I need to manually change the en.json file and add diaper sprite every time you do an update to the mod?)

**Reply by zippity21** [reply]
Not if you share it :)

I don't know if you're familiar at all with Git, but by having this project on GitHub, it allows anybody to contribute.

I'd be happy to help with specifics, but the general idea is that you make a "clone" of this code on your computer, create a  "branch" that is an offshoot, then "commit" your changes (in this case the sprite and en.json files), then you "push" the branch back to GitHub, and then initiate a "pull request" to get it accepted into the main branch (which is where I release from). You do all but the last part. As the maintainer, I have to approve the pull request. This allows for a certain level of quality control. And if this were a larger project, there would be multiple reviewers.

**_If you don't feel like learning all that, you could probably share those files here in this discussion (it supports drag and drop) and I can do everything from there._**

However, the advantage of doing it yourself is that you have control of your own branch, and you get the credit for changes, since Git keeps a history of authors.

Edit, adding resources:
To install Git: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
To contribute to a project on GitHub: https://www.dataschool.io/how-to-contribute-on-github/

**Comment by Kerie70** [comment]
I'll try and get the github thing set up in the future, for now I'll just post the two minor changes I made here, my sprites are kinda meh but I've never done pixel art before so it is what it is. 
![sprites](https://user-images.githubusercontent.com/93156174/139736758-43ab2761-6e69-4aef-8d0e-7eb18ef5f040.png)

Edit: removed old diaper text

**Reply by zippity21** [reply]
Way better than I could have done. Thank you, I'll get these added to main. I won't do a release yet, since I'd like to accumulate a few changes between releases (otherwise people may feel inundated). But rest assured that it'll be in the next release.

**Reply by Kerie70** [reply]
I uploaded the wrong text for the diaper, this one is the correct one and has minor changes.  Sorry about that.
[heartdiaper.txt](https://github.com/zippity21/Stardew_Valley_Regression_Mod/files/7458290/heartdiaper.txt)
